src/services/realtime/ably/index.ts

```diff
@@ -1010,42 +1014,47 @@ export default class AblyRealtimeService extends RealtimeService implements Ably
     }, timeToWait);
   };
 
+  private async throttleCallback(myPresenceParam: AblyParticipant | Ably.Types.PresenceMessage) {
+    const usedSlots: Ably.Types.PresenceMessage[] = [];
+    let myPresence = myPresenceParam;
+    this.supervizChannel.presence.get((err, members) => {
+      members.forEach((member) => {
+        if (member.connectionId === myPresence.connectionId) {
+          myPresence = member;
+        }
+        if (
+          member.connectionId !== myPresence.connectionId &&
+          member.data.slotIndex !== undefined
+        ) {
+          usedSlots.push(member.data.slotIndex);
+        }
+      });
+    });
+
+    if (
+      this.myParticipant.data.slotIndex === undefined ||
+      usedSlots.includes(this.myParticipant.data.slotIndex)
+    ) {
+      this.findSlotIndex(myPresence);
+    } else {
+      // confirm slot and propagate
+      const roomProperties = await this.fetchRoomProperties();
+      this.updateRoomProperties(roomProperties);
+
+      this.presenceSlotsInfosObserver.publish({
+        slotIndex: this.myParticipant.data.slotIndex,
+        id: this.myParticipant.data.id,
+      });
+    }
+  }
```

Comment:
It's a good point, but I think the original approach is better; in general, we leave the callback directly inside the throttle because it's more readable, and it also reduces the risk of the method being called directly without the throttle.

```suggestion
```